blocks/api/factory.js

```diff
@@ -71,15 +71,17 @@ export function createBlock( name, blockAttributes = {}, innerBlocks = [] ) {
  *
  * @return {Object} A cloned block.
  */
-export function cloneBlock( block, mergeAttributes = {}, innerBlocks = block.innerBlocks ) {
+export function cloneBlock( block, mergeAttributes = {}, innerBlocks ) {
 	return {
 		...block,
 		uid: uuid(),
 		attributes: {
 			...block.attributes,
 			...mergeAttributes,
 		},
-		innerBlocks,
+		innerBlocks: innerBlocks ?
+			innerBlocks :
+			( block.innerBlocks || [] ).map( ( innerBlock ) => cloneBlock( innerBlock ) ),
```

Comment:
When will `innerBlocks` be falsey?

Reply:
Only if the block was wrongly changed/created, but yes we should not be "defending" against this case, I removed this check.